data/campaigns/tutorial/scenarios/02_Tutorial_part_2.cfg

```
@@ -894,6 +894,13 @@ A full list of abilities and weapons specials, along with traits, may be found i
         [/filter_condition]
 
         {TALK_ABOUT Thrag ( _ "You are close to killing their leader! The unit who finishes him will gain 16 experience points because he is second level. Choose your attacking unit carefully!")}
+
+        [message]
+            speaker=narrator
+            caption= _ "You have almost finished the tutorial"
+            image=wesnoth-icon.png
+            message= _ "This tutorial will conclude after you defeat Thrag. In a campaign you may have ten or even twenty scenarios to build your army; the rookie-level campaigns have hints about choosing which unit should get those 16 experience points."
```

I don't think the reference to rookie campaigns belongs here. This is the `attack end` event. The player hasn't played those campaigns, doesn't know what the hints say, and can't find out (not during the scenario). There is already a reference to the rookie campaigns in the `victory` event, so I would remove it.

Then I think the remaining two clauses should be restored to the original order:

```suggestion
            message= _ "In a campaign you may have ten or even twenty scenarios to build your army, but this tutorial will conclude after you defeat Thrag."
```

---

I'm happy to do that, but think "Choose your attacking unit carefully!" should be removed from the line before this one, because there's neither hints about which unit to choose nor any reward for thinking about it (because the scenario ends immediately).

---

Ping?

---

Real life got a bit too busy recently.

---

No worries! Are you still planning to get around to this or would you prefer that someone else would take over?

---

My new revision is to have Galdrad say "You are close to killing their leader! Because he is second level, the unit who finishes him will gain 16 experience points. In a campaign, it's important to choose which units you want to advance, and give those units the best chances to gain experience."

I'll push this and will have time to do more changes in the next couple of days, but I'm also happy if someone else takes over.